storage/innobase/include/buf0buf.h

```diff
@@ -1746,7 +1748,12 @@ class buf_pool_t
     /** @return the index of an array element */
     ulint calc_hash(ulint fold) const { return calc_hash(fold, n_cells); }
     /** @return raw array index converted to padded index */
-    static ulint pad(ulint h) { return 1 + (h / ELEMENTS_PER_LATCH) + h; }
+    static ulint pad(ulint h)
+    {
+      ulint latches = h / ELEMENTS_PER_LATCH;
+      ulint empty_slots = latches * EMPTY_SLOTS_PER_LATCH;
+      return 1 +  latches + empty_slots + h;
+    }
```

Comment:
I will have to check that various compilers will optimize away the `empty_slots=0` on systems with 64-bit cache line size. I expect it to, but the oldest compiler that we support (GCC 4.8.5) is pretty old.

Formatting nitpick: remove space before each `=`, and the double space in `1 + latches`.

Reply:
The computation of `empty_slots` will be completely optimized away already at `-O1` (or GCC `-Og`) when `EMPTY_SLOTS_PER_LATCH` is 0. I tested with various GCC versions starting from 4.8.1, and I also tested `clang` 4.0.0. At `-O0` each compiler will generate some extra code.

Reply:
formating fixed.